### ISSUE TITLE
Dev: How to find out which URL the response is from?

### ISSUE THREAD

**Issue by pllim** [astropy/astroquery#850]
I am trying to follow http://astroquery.readthedocs.io/en/latest/index.html#developer-documentation in order to transplant Cone Search from Astropy to here. In Cone Search, I need to know which service URL that is attached to a response. However, in your API, the arg parsing and the response handling are separated into two methods. Your URL attribute is a list of mirrors (or in my case, a list of different services to try in order), then you pass this whole thing into the `GET` stuff, and then suddenly I have response from one of them, but which one? How do I find out?

**Comment by keflavich** [comment]
The default is to just use the first in the list in the `configuration` for each module.  

You're saying that, in your case, you want to give a list of URLs and try each of them in turn?  That's fine, but it should be implemented specifically for your module.

My usual approach for this sort of thing is to store the response in `self._last_response`, then you should be able to get the url with `self._last_response.url` (where `_last_response` is the object returned from the `request`)

But, I may not have answered your question because I'm not sure we're speaking entirely the same language yet.  If I got it wrong, could you provide some example pseudocode?

**Comment by pllim** [comment]
Let's use http://astroquery.readthedocs.io/en/latest/api.html#outline-of-an-example-module for this discussion. I tried running some existing services but it is not clear to me how this works exactly.

There is a list in `conf.Server` but somehow `QC.server` is only a string. So, somewhere behind the scene, `QC` went through the list and picked one. Is this a one time thing at initialization? Or does `QC` go through the list every time during the actual query?

Also, are you saying that the `response` from `QC.query_X_async()` *always* has a `.url` attribute attached to it? Does that URL include `args` or just the "base" URL?

**Comment by keflavich** [comment]
Yes, `response` will always have a url, but it is just the base url.

for server selection, I have to dig more...
